src/atdf/peripheral.rs

```diff
@@ -9,13 +9,26 @@ pub fn parse_list(
 ) -> crate::Result<Vec<chip::Peripheral>> {
     let mut peripherals = vec![];
 
-    for module in el.children.iter() {
-        module.check_name("module")?;
+    for module in el.children.iter().filter_map(|node| {
+        let module = node.as_element().filter(|e| e.name == "module");
+
+        if module.is_none() {
+            log::warn!("Unhandled module element: {module:?}");
+        }
+
+        module
+    }) {
```

Comment:
It seems your new code uses this pattern quite a lot.  Maybe it makes sense to create a new method in `ElementExt` for it?  Something like

```rust
fn iter_children_with_name(name: &str, warning_parent_name: Option<&str>) -> ... {
    self.children.iter().filter_map(|node| {
        let child = node.as_element().filter(|e| e.name == name);
        if let Some(parent_name) = warning_parent_name {
            if child.is_none() {
                log::warn!("Unhandled child element in <{parent_name}>: {child:?}");
            }
        }
        child
    })
}
```